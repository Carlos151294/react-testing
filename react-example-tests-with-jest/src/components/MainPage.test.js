import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
  const mockProps = {
    robots: [],
    searchField: '',
    isPending: false,
    onSearchChange: jest.fn(),
    onRequestRobots: jest.fn()
  };

  wrapper = shallow(<MainPage {...mockProps} />);
});

it('renders MainPage Component', () => {
  expect(wrapper).toMatchSnapshot();
});

describe('Test filterRobots method', () => {
  it('filters robots correctly with empty robots array', () => {
    expect(wrapper.instance().filterRobots()).toEqual([]);
  });

  it('filters robots correctly with a robot matching the searchField', () => {
    const mockProps2 = {
      robots: [{
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
      }],
      searchField: 'john',
      isPending: false,
      onSearchChange: jest.fn(),
      onRequestRobots: jest.fn()
    };
  
    const wrapper2 = shallow(<MainPage {...mockProps2} />);
  
    expect(wrapper2.instance().filterRobots()).toEqual([{
      id: 3,
      name: 'John',
      email: 'john@gmail.com'
    }]);
  });

  it('filters robots correctly for non-matching robots', () => {
    const mockProps3 = {
      robots: [{
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
      }],
      searchField: 'a',
      isPending: false,
      onSearchChange: jest.fn(),
      onRequestRobots: jest.fn()
    };
  
    const wrapper3 = shallow(<MainPage {...mockProps3} />);
  
    expect(wrapper3.instance().filterRobots()).toEqual([]);
  });
});
