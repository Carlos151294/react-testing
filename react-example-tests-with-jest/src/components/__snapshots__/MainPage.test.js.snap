// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage Component 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox
    searchChange={[MockFunction]}
  />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
